security/access_denied_handler.rst

```diff
@@ -70,3 +70,8 @@ configure it under your firewall:
 
 That's it! Any ``AccessDeniedException`` thrown by code under the ``main`` firewall
 will now be handled by your service.
+
+.. note::
+
+    The ``AccessDeniedHandler`` will not be invoked if an anonymous user is trying to
+    access a protected resource. That will be an ``InsufficientAuthenticationException``.
```

Comment:
I don't think the second sentence is very relevant (that's an implementation detail, required to run entry points). What about something like "In that case, the authentication entry point will be called" (I realize that we do not explain "entry point" anywhere in the docs 😞, but that'll need to change for #13605 anyways).